.github/workflows/cypress_tests.yml

```diff
@@ -0,0 +1,83 @@
+name: Run cypress happy path test
+
+on:   
+  pull_request:
+    branches:
+      - integration
+
+jobs:
+  build:
+    timeout-minutes: 10
+    runs-on: ubuntu-20.04
+    env:
+      JWT_INVITE_SECRET: ${{secrets.JWT_INVITE_SECRET}}
+      JWT_RESET_SECRET: ${{secrets.JWT_RESET_SECRET}}
+      JWT_SCHEDULER_SECRET: ${{secrets.JWT_SCHEDULER_SECRET}}
+      JWT_SECRET: ${{secrets.JWT_SECRET}}
+      TEST_USER: ${{secrets.TEST_USER}}
+      TEST_USER_ID: ${{secrets.TEST_USER_ID}}
+      DEV_DATABASE_HOST: localhost
+      DEV_DATABASE: pg-litefarm
+      DEV_DATABASE_USER: postgres
+      DEV_DATABASE_PASSWORD: postgres
+      GOOGLE_API_KEY: ${{secrets.VITE_GOOGLE_MAPS_API_KEY}}
+      VITE_GOOGLE_MAPS_API_KEY: ${{secrets.VITE_GOOGLE_MAPS_API_KEY}}
+      VITE_WEATHER_API_KEY: ${{secrets.VITE_WEATHER_API_KEY}}
+      
+    strategy:
+      matrix:
+        node-version: [17.5.0]
+
+    steps:
+      - uses: ikalnytskyi/action-setup-postgres@v3
+        with:
+          username: ${{env.DEV_DATABASE_USER}}
+          password: ${{env.DEV_DATABASE_PASSWORD}}
+          database: ${{env.DEV_DATABASE}}
+          port: 5432
+
+      - uses: actions/checkout@v2
+
+      - name: Use Node.js ${{ matrix.node-version }}
+        uses: actions/setup-node@v1
+        with:
+          node-version: ${{ matrix.node-version }}
+
+      - name: Run API in background
+        working-directory: packages/api
+        run: |
+          npm ci && cd ../shared && npm ci
+          cd ../api && npm run migrate:dev:db
+          npm run start &
+        env:
+          PORT: 5000
+          
+      - name: Install pnpm
+        uses: pnpm/action-setup@v2
+        with:
+          version: latest
+
+      - name: Run and test app
+        env:   
+          VITE_GOOGLE_MAPS_API_KEY: AIzaSyC8UBxTVXeVnFGcZnvhTBY19rpkg6kEL3I
```

Comment:
@MwayaB ... If this is still the temporary key I shared for testing the one day it could be the reason you had trouble with GMaps? If however this is LiteFarms key you will need to generate a new one and add it to secrets since this is exposed.